core/spi.c

```diff
@@ -74,6 +75,10 @@ spi_init(void)
   PIN_SET(S1D15G10_CS);
 #endif
 
+#ifdef CAN_SUPPORT
```

Comment:
Please do not use such a generic symbol for a chip specific feature. I suggest using MCP2515_SUPPORT instead of CAN_SUPPORT.